fix(HMSPROV-392): Check if image is an original or a shared one

adiabramovitch opened the pull request:
Images can be shared to different regions in Image Builder. 
shared images are not visible under /compose endpoint in IB but under /clone.
Therefore if an image is not visible under /compose,  before erroring out we should search for the image under /clone. 
Only if the image is not preset neither under compose nor clone we should error out with the message that the image is not available.

ezr-ondrej reviewed (commented):
Few nits, but nothing major.
Given we are calling Image builder twice now, I'd think about adding OpenTelemetry tracing around it and measure those calls in their own span, but that should be (if we decide to do it) a follow up anyway :)

adiabramovitch commented:
> Few nits, but nothing major. Given we are calling Image builder twice now, I'd think about adding OpenTelemetry tracing around it and measure those calls in their own span, but that should be (if we decide to do it) a follow up anyway :)

Fixed and added :)

lzap commented on internal/clients/http/image_builder/image_client.go:
```diff
@@ -105,9 +105,26 @@ func (c *ibClient) GetGCPImageName(ctx context.Context, composeID string) (strin
+	if err != nil {
+		cloneResp, err := c.checkClone(ctx, composeID)
+		if err != nil {
+			return nil, fmt.Errorf("could not find image neither in compose nor in clones %w", err)
```
Mind adding colon, error could be misleading without it:

```suggestion
			return nil, fmt.Errorf("could not find image neither in compose nor in clones: %w", err)
```

lzap commented on internal/clients/http/image_builder/image_client.go:
```diff
@@ -123,8 +140,35 @@ func (c *ibClient) fetchImageStatus(ctx context.Context, composeID string) (*Upl
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			return nil, fmt.Errorf("fetch image status call: %w", http.ComposeNotFoundErr)
```
This is `checkClone` so:

```suggestion
			return nil, fmt.Errorf("fetch image status call: %w", http.CloneNotFoundErr)
```

lzap reviewed (changes requested):
Two error handling comments.

lzap reviewed (approved):
Leaving this on @amirfefer since he is working on this IIRC.

amirfefer reviewed (approved):
Looks good, tested multiple image's clones 
Thanks @adiabramovitch 👍
